src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -155,38 +155,31 @@ public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
             if (padding == null)
                 throw new ArgumentNullException("padding");
 
-            Interop.libcrypto.OpenSslRsaPadding openSslPadding;
-
+            bool useOaepPadding;
             if (padding == RSAEncryptionPadding.Pkcs1)
             {
-                openSslPadding = Interop.libcrypto.OpenSslRsaPadding.RSA_PKCS1_PADDING;
+                useOaepPadding = false;
```

Comment:
Sure, but when we get there, we might very well do it here, no? Is the argument that the manual stuff we have to do will require different shimmed functions rather than some extra C++ here? Even then, enums are easier to read than bools and there's always the possibility that we get a future OpenSSL version that does make it easy. It would be unfortunate IMHO, to add a new function just to account for more than two values. As a compromise, you could error out if neither 0 nor 1 is passed and compatibly switch bool -> int32_t marshalling to enum -> enum marshalling. I'd just make the enum, though, there's no need for any static_asserts or conversions, just switch over the input here. I'm not understanding the downside.